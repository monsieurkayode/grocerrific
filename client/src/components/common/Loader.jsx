import React from 'react';
import { number } from 'prop-types';
import PreloaderIcon from 'react-preloader-icon';
import Puff from 'react-preloader-icon/loaders/Puff';

/**
 *
 * @function Loader
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const Loader = ({ size }) => (
  <div className="loader">
    <PreloaderIcon
      loader={Puff}
      size={size}
      strokeWidth={12}
      strokeColor="#4d75a1"
      duration={1200}
    />
  </div>
);

Loader.propTypes = {
  size: number.isRequired
};

export default Loader;
