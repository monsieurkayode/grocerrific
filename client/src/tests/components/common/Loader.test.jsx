import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../../components/common/Loader';

describe('Loader Component', () => {
  const wrapper = shallow(<Loader size={20} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a single root div with "loader" class', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('.loader').length).toBe(1);
  });

  it('contains a PreloaderIcon Component', () => {
    expect(wrapper.find('PreloaderIcon').length).toBe(1);
  });
});
