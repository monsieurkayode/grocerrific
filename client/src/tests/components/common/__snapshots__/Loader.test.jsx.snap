// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders correctly 1`] = `
<div
  className="loader"
>
  <PreloaderIcon
    className=""
    duration={1200}
    loader={[Function]}
    size={20}
    strokeColor="#4d75a1"
    strokeWidth={12}
    style={Object {}}
    unit="px"
  />
</div>
`;
